### PR TITLE
Bump py-slepc4py version, update url

### PR DESCRIPTION
**Pull request by s-sajid-ali**

### CODE REVIEW

**Comment by adamjstewart** [comment]
@joseeroman @balay I see that you're listed as maintainers for the `slepc` package. Would you also like to maintain `py-slepc4py`? Do you have any comments on this PR?

File: var/spack/repos/builtin/packages/py-slepc4py/package.py [hunk 1]
```diff
@@ -10,10 +10,11 @@ class PySlepc4py(PythonPackage):
...
     """
 
-    homepage = "https://bitbucket.org/slepc/slepc4py"
-    url      = "https://bitbucket.org/slepc/slepc4py/get/3.10.0.tar.gz"
-    git      = "https://bitbucket.org/slepc/slepc4py.git"
+    homepage = "https://gitlab.com/slepc/slepc4py"
+    url      = "https://gitlab.com/slepc/slepc4py/-/archive/3.12.0/slepc4py-3.12.0.tar.gz"
+    git      = "https://gitlab.com/slepc/slepc4py.git"
 
+    version('3.12.0', sha256='d8c06953b7d00f529a9a7fd016dfa8efdf1d05995baeea7688d1d59611f424f7')
```
Comment (balay) [inline var/spack/repos/builtin/packages/py-slepc4py/package.py]: @s-sajid-ali might aswell add an entry for 3.11.0?

**Comment by balay** [comment]
> Are the checksums for old releases still the same?

It should be the same. The repos moved from bitbucket to gitlab

We should also update slepc package with this change.

**Comment by balay** [comment]
> @joseeroman @balay I see that you're listed as maintainers for the `slepc` package. Would you also like to maintain `py-slepc4py`? Do you have any comments on this PR?

Yes - and @dalcinl

**Comment by balay** [comment]
I pushed my fixes to this PR

**Comment by s-sajid-ali** [comment]
@balay : Thanks for enhancing the PR !

**Pull request merged**